app/src/main/java/diemmy/edu/recycler/MainActivity.java

```diff
@@ -2,12 +2,51 @@
 
 import android.support.v7.app.AppCompatActivity;
 import android.os.Bundle;
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
 
-public class MainActivity extends AppCompatActivity {
+import java.util.ArrayList;
+import java.util.List;
+
+public class MainActivity extends AppCompatActivity implements ViewAdapter.IViewAdapter {
+    private ViewAdapter viewAdapter;
```

Comment:
2.1 Java Style Rules

Field names need to be defined at first and follow this conventions.

Non-public, non-static field names start with m.
Static field names start with s.
Other fields start with a lower case letter.
Public static final fields (constants) are ALL_CAPS_WITH_UNDERSCORES.